packages/modules/npm.py

```diff
@@ -1,21 +1,44 @@
 import requests
+import semver
 
 from packages.models import VersionedPackage
 
 NPM_REGISTRY_URL = "https://registry.npmjs.org"
 
 
-def get_versioned_package(name: str, version: str) -> VersionedPackage:
-    url = f"{NPM_REGISTRY_URL}/{name}/{version}"
+def get_package(name: str, range: str) -> VersionedPackage:
```

Comment:
Let's add caching (as the ticket required). I've set the max caching size to 1,000 (over the default 128), I think that should give a decent performance boost for large dependency trees, and shouldn't use too much memory. Please check.

```suggestion
@lru_cache(maxsize=1000)
def get_package(name: str, range: str) -> VersionedPackage:
```